blamo/lib/main.dart

```diff
@@ -1,142 +1,231 @@
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+/* Builds the initial page for the user
+*  Using Scaffolding to follow a fairly traditional layout for familiarity
+*  The drawer needs to house the other pages our team is working on (Settings, Export, etc)
+*/
+class _HomePageState extends State<HomePage> {
+  static const routeName = '/';
+  StateData currentState = StateData(routeName);
 
-class MyApp extends StatelessWidget {
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
+    if(ModalRoute.of(context).settings.arguments != null) {
+      currentState = ModalRoute
+          .of(context)
+          .settings
+          .arguments;
+      currentState.currentRoute = '/'; // Recieves StateData objects from Navbar
+    }
+
+    /* Scaffolding constructor is as follows, and can be filled out of order using the precursor of
+    * X: new Y(),
+    *
+    * Scaffold(key, appBar, Widget body, Widget floatingActionButton, FloatingActionButtonLocation
+    * floatingActionButtonLocation, FloatingActionButtonAnimator floatingActionButtonAnimator,
+    * list<Widget> persistentFooterButtons, Widget drawer, Widget endDrawer,
+    * Widget bottomNavigationBar, Widget bottomSheet, Color backgroundColor,
+    * bool resizeToAvoidBottomPadding, bool resizeToAvoidBottomInset, bool primary: true,
+    * DragStartBehavior drawerDragStartBehavior: DragStartBehavior.start,
+    * bool extendBody: false, bool extendBodyBehindAppBar: false, Color drawerScrimColor,
+    * double drawerEdgeDragWidth )
+    *
+    * */
+    return new Scaffold(
+      drawer: Drawer(
+        child:SideMenu(currentState),
+      ),
+      appBar: new AppBar(
+          title: new Text("Home"),
+          actions: <Widget>[
+
+          ],
+          backgroundColor: Colors.deepOrange
+      ),
+      body: GridView.count(
+        padding: const EdgeInsets.all(20),
+        crossAxisSpacing: 2,
+        mainAxisSpacing: 10,
+        crossAxisCount: 2,
+        children: List.generate(currentState.randomNumber + 1, (index) {
+          String toReturn;
+          int colorVal = index*100;
+          if(index >= 9){
+            colorVal = 800;
+          }
+          if(index == 0){
+            toReturn = '+';
```

Comment:
Is this index 0 doc a placeholder? If it is, something we could do give it the add doc functionality that the button will have and get rid of the button. Might be a case of "if it aint broke" though, just a thought that's not that important for this pr.